### PR TITLE
Implement access for Google Cloud buckets with Requester Pays

### PR DESCRIPTION
**Pull request by emi80**

### CODE REVIEW

File: modules/nf-google/src/main/nextflow/cloud/google/util/GsPathFactory.groovy [hunk 1]
```diff
@@ -30,14 +33,29 @@ import nextflow.file.FileSystemPathFactory
...
+            final builder = CloudStorageConfiguration.builder()
+            if (config.useRequesterPays) {
+                builder.userProject(config.project)
+            }
+            storageConfig = builder.build()
+        } catch(NullPointerException) {
+            storageConfig = CloudStorageConfiguration.DEFAULT
+        }
+    }
+
```
Comment (pditommaso) [inline modules/nf-google/src/main/nextflow/cloud/google/util/GsPathFactory.groovy:50]: Does this need to be done for each file? Could not the config created just once using [lazy initialization](https://github.com/nextflow-io/nextflow/blob/a5a6a1fc8f76a9a7ebbe98cc94b5356f2a23bdec/modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy#L47-L58)? 

Also the `catch(NullPointerException)` smell bad, what's the need for it?

File: modules/nf-google/src/main/nextflow/cloud/google/lifesciences/GoogleLifeSciencesFileCopyStrategy.groovy [hunk 2]
```diff
@@ -72,13 +79,13 @@ class GoogleLifeSciencesFileCopyStrategy extends SimpleFileCopyStrategy {
...
             if(storePathIsDir) {
-                stagingCommands << "gsutil -m -q cp -R $escapedStoreUri/ $localTaskDir".toString()
+                stagingCommands << "$gsutilPrefix cp -R $escapedStoreUri/ $localTaskDir".toString()
                 //check if we need to move the directory (gsutil doesn't support renaming directories on copy)
                 if(parent || !storePath.toString().endsWith(stageName)) {
                     stagingCommands << "mv $localTaskDir/${Escape.path(storePath.name)} $localTaskDir/$escapedStageName".toString()
                 }
             } else {
-                stagingCommands << "gsutil -m -q cp $escapedStoreUri $localTaskDir/$escapedStageName".toString()
+                stagingCommands << "$gsutilPrefix cp $escapedStoreUri $localTaskDir/$escapedStageName".toString()
```
Comment (pditommaso) [inline modules/nf-google/src/main/nextflow/cloud/google/lifesciences/GoogleLifeSciencesFileCopyStrategy.groovy:88]: We need unit tests for this part.

File: modules/nf-google/src/main/nextflow/cloud/google/util/GsPathFactory.groovy (hunk 1 above)
Comment (emi80) [reply modules/nf-google/src/main/nextflow/cloud/google/util/GsPathFactory.groovy:53]: Changed in aa5550f

File: modules/nf-google/src/main/nextflow/cloud/google/lifesciences/GoogleLifeSciencesFileCopyStrategy.groovy (hunk 2 above)
Comment (emi80) [reply modules/nf-google/src/main/nextflow/cloud/google/lifesciences/GoogleLifeSciencesFileCopyStrategy.groovy:88]: Added test in aa5550f

File: modules/nf-google/src/main/nextflow/cloud/google/util/GsPathFactory.groovy (hunk 1 above)
Comment (pditommaso) [reply modules/nf-google/src/main/nextflow/cloud/google/util/GsPathFactory.groovy:53]: Maybe it should an `IllegalStateException` if the session is missing.

**Comment by emi80** [comment]
Updated Google Life Sciences options in docs

**Comment by emi80** [comment]
Moved `enableRequesterPaysBuckets` option to `google` scope in bc9db8d

**Pull request merged**